File: data/CodeLlama/readable_format_HPEC/fortran/file240.f90
subroutine kernel_adi(tsteps, n, x, a, b)
implicit none

DATA_TYPE, dimension(n, n) :: a
DATA_TYPE, dimension(n, n) :: x
DATA_TYPE, dimension(n, n) :: b
integer :: n, tsteps
integer :: i1, i2, t

!$pragma scop
do t = 1, _PB_TSTEPS
do i1 = 1, _PB_N
do i2 = 2, _PB_N
x(i2, i1) = x(i2, i1) - ((x(i2 - 1, i1) * a(i2, i1)) / &
b(i2 - 1, i1))
b(i2, i1) = b(i2, i1) - ((a(i2, i1) * a(i2, i1)) / &
b(i2 - 1, i1))
end do
end do

do i1 = 1, _PB_N
x(_PB_N, i1) = x(_PB_N, i1) / b(_PB_N, i1)
end do

do i1 = 1, _PB_N
do i2 = 1, _PB_N - 2
x(_PB_N - i2, i1) = (x(_PB_N - i2, i1) - &
(x(_PB_N - i2 - 1, i1) * &
a(_PB_N - i2 - 1, i1))) / &
b(_PB_N - i2 - 1, i1)
end do
end do

do i1 = 2, _PB_N
do i2 = 1, _PB_N
x(i2, i1) = x(i2, i1) - x(i2, i1 - 1) * a(i2, i1) / &
b(i2, i1 - 1)
b(i2, i1) = b(i2, i1) - a(i2, i1) * a(i2, i1) / &
b(i2, i1 - 1)

end do
end do

do i2 = 1, _PB_N
x(i2, _PB_N) = x(i2, _PB_N) / b(i2, _PB_N)
end do

do i1 = 1, _PB_N - 2
do i2 = 1, _PB_N
x(i2, _PB_N - i1) = (x(i2, _PB_N - i1) - &
x(i2, _PB_N - i1 - 1) * &
a(i2, _PB_N - i1 - 1)) / &
b(i2, _PB_N - i1)
end do
end do
end do
!$pragma endscop
end subroutine

File: data/gpt-4o/readable_format_HPEC/cpp/file240.cpp
void kernel_adi(int tsteps, int n, float** x, float** a, float** b){
 for (int t = 0; t < tsteps; t++) {
 for (int i1 = 0; i1 < n; i1++) {
 for (int i2 = 1; i2 < n; i2++) {
 x[i2][i1] -= (x[i2 - 1][i1] * a[i2][i1]) / b[i2 - 1][i1];
 b[i2][i1] -= (a[i2][i1] * a[i2][i1]) / b[i2 - 1][i1];
 }
 }
 for (int i1 = 0; i1 < n; i1++) {
 x[n - 1][i1] /= b[n - 1][i1];
 }
 for (int i1 = 0; i1 < n; i1++) {
 for (int i2 = 0; i2 < n - 2; i2++) {
 x[n - i2 - 2][i1] = (x[n - i2 - 2][i1] - (x[n - i2 - 3][i1] * a[n - i2 - 3][i1])) / b[n - i2 - 3][i1];
 }
 }
 for (int i1 = 1; i1 < n; i1++) {
 for (int i2 = 0; i2 < n; i2++) {
 x[i2][i1] -= x[i2][i1 - 1] * a[i2][i1] / b[i2][i1 - 1];
 b[i2][i1] -= a[i2][i1] * a[i2][i1] / b[i2][i1 - 1];
 }
 }
 for (int i2 = 0; i2 < n; i2++) {
 x[i2][n - 1] /= b[i2][n - 1];
 }
 for (int i1 = 0; i1 < n - 2; i1++) {
 for (int i2 = 0; i2 < n; i2++) {
 x[i2][n - i1 - 2] = (x[i2][n - i1 - 2] - (x[i2][n - i1 - 3] * a[i2][n - i1 - 3])) / b[i2][n - i1 - 2];
 }
 }
 }
}
